sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/RequiresStableInputIT.java

```diff
@@ -0,0 +1,167 @@
+      }
+    }
+
+    private static void writeTextToFileSideEffect(String text, String filename) throws IOException {
+      ResourceId rid = FileSystems.matchNewResource(filename, false);
+      WritableByteChannel chan = FileSystems.create(rid, "text/plain");
+      chan.write(ByteBuffer.wrap(text.getBytes(Charset.defaultCharset())));
+      chan.close();
+    }
+  }
+
+  private static void runRequiresStableInputPipeline(RequiresStableInputTestOptions options) {
+    Pipeline p = Pipeline.create(options);
+
+    PCollection<String> singleton = p.apply("CreatePCollectionOfOneValue", Create.of(VALUE));
+    singleton
+        .apply("Single-PairWithRandomKey", MapElements.via(new PairWithRandomKeyFn()))
+        .apply(
+            "Single-MakeSideEffectAndThenFail",
+            ParDo.of(new MakeSideEffectAndThenFailFn(options.getParDoSingleOutputPrefix())));
+    singleton
+        .apply("Multi-PairWithRandomKey", MapElements.via(new PairWithRandomKeyFn()))
+        .apply(
+            "Multi-MakeSideEffectAndThenFail",
+            ParDo.of(new MakeSideEffectAndThenFailFn(options.getParDoMultiOutputPrefix()))
+                .withOutputTags(new TupleTag<>(), TupleTagList.empty()));
+
+    p.run().waitUntilFinish();
+  }
+
+  @BeforeClass
+  public static void setup() {
+    PipelineOptionsFactory.register(TestPipelineOptions.class);
+  }
+
+  /**
+   * Option for RequiresStableInputIT.
+   *
+   * <p>Define the output prefixes for tests on {@link ParDo.SingleOutput} and {@link
+   * ParDo.MultiOutput}.
+   */
+  public interface RequiresStableInputTestOptions extends TestPipelineOptions {
+    @Description("Prefix of the output files for testing ParDo.SingleOutput")
+    @Required
+    String getParDoSingleOutputPrefix();
+
+    void setParDoSingleOutputPrefix(String value);
+
+    @Description("Prefix of the output files for testing ParDo.MultiOutput")
+    @Required
+    String getParDoMultiOutputPrefix();
+
+    void setParDoMultiOutputPrefix(String value);
+  }
+
+  @Test
+  public void testParDoRequiresStableInput() {
+    RequiresStableInputTestOptions options =
+        TestPipeline.testingPipelineOptions().as(RequiresStableInputTestOptions.class);
+    ResourceId dir =
+        FileSystems.matchNewResource(options.getTempRoot(), true)
+            .resolve(
+                String.format("requires-stable-input-%tF-%<tH-%<tM-%<tS-%<tL", new Date()),
+                StandardResolveOptions.RESOLVE_DIRECTORY);
+
+    options.setParDoSingleOutputPrefix(
+        dir.resolve("pardo-single-output", StandardResolveOptions.RESOLVE_DIRECTORY)
+            .resolve("key-", StandardResolveOptions.RESOLVE_FILE)
+            .toString());
+    options.setParDoMultiOutputPrefix(
+        dir.resolve("pardo-multi-output", StandardResolveOptions.RESOLVE_DIRECTORY)
+            .resolve("key-", StandardResolveOptions.RESOLVE_FILE)
+            .toString());
+    options.setOnSuccessMatcher(
```

Comment:
Your overwriting the on success matcher by the line below, you want to use [Matchers#allOf](http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#allOf(org.hamcrest.Matcher...))